docs/tutorial/csrman_design.md

```diff
@@ -1,21 +1,119 @@
+
+![image](https://github.com/KULeuven-MICAS/snitch_cluster/assets/26665295/b9c4758a-ad20-4e3b-86f8-5dc7b649df09)
+
+In our [SNAX ALU accelerator](./accelerator_design.md), we have 2 main configuration RW registers: the mode and the data length to process. The 3rd RW register is the start that sends the configured settings to the accelerator data path. It is only when you assert the LSB of the last RW register (or the start register) that the CSR manager transacts all configurations to the accelerator.
+
+When the accelerator is not available to accept configuration transactions, the CSR holds the last transaction until it is successful. The timing diagram below shows this:
+
+![image](https://github.com/KULeuven-MICAS/snitch_cluster/assets/26665295/bb31e9e5-4f80-4acc-b9ee-72f6866bec13)
+
+The CSR manager is useful for double buffering. For example, when an accelerator is already processing an old configuration, the Snitch core can already configure new settings. This does not overwrite the old configuration. This is particularly useful in hiding CSR setup cycles because we can set up configurations while the accelerator is running.
+
+The RO registers are read-only registers that the CSR manager can read from. These are mostly used for monitoring purposes like status or performance counters. The addresses of the RO registers are immediately after the RW registers. For example, the register table in [Accelerator Design](./accelerator_design.md) is copied below. Observe that all RO registers are after the RW registers.
+
+
+|  register name  |  register addr  |   type  |                   description                       |
+| :-------------: | :-------------: | :-----: |:--------------------------------------------------: |
+|    mode         |       0         |   RW    | Operating modes: 0 - add, 1 - sub, 2 - mul, 3 - XOR |
+|    length       |       1         |   RW    | Number of elements to process                       |
+|    start        |       2         |   RW    | Set 1 to LSB only to start the accelerator          |
+|    busy         |       3         |   RO    | Busy status. 1 - busy, 0 - idle                     |
+|  perf. counter  |       4         |   RO    | Performance counter indicating number of cycles     |
 
-# Detailed Block Diagram of CSR Manager
 
 
 # Configuring the Generated CSR Manager
 
+You only need to specify the number of RW and RO registers of the CSR manager in the `snax-alu.hjson` configuration file. The snippet below shows the `snax_acc_cfg` dictionary:
+
+```json
+snax_acc_cfg: {
+    snax_acc_name: "snax_alu"
+    snax_tcdm_ports: 16,
+    snax_num_rw_csr: 3,
+    snax_num_ro_csr: 2,
+    snax_streamer_cfg: {$ref: "#/snax_alu_streamer_template" }
+},
```

Comment:
Why do we use red for the cfg example here? It looks a little inconsistent with other parts

Reply:
it's a wrong rendering of the `json` file, it should be fixed already with Josse's suggestion 😄 I should've used `hjson` instead.